pkg/reconciler/broker/controller.go

```diff
@@ -111,17 +112,17 @@ func NewController(
 	})
 
 	// Reconcile Broker (which transitively reconciles the triggers), when Subscriptions
-	// to those triggers change.
+	// that I own are changed.
 	subscriptionInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
 		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
 		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 	})
 
-	// Reconcile Broker (which transitively reconciles the triggers), when subscriptions
-	// to those triggers change.
+	// Reconcile trigger (by enqueuing the broker specified in the label) when subscriptions
+	// of triggers change.
 	subscriptionInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
-		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+		FilterFunc: pkgreconciler.LabelExistsFilterFunc("eventing.knative.dev/broker"),
+		Handler:    controller.HandleAll(impl.EnqueueLabelOfNamespaceScopedResource("" /*any namespace*/, "eventing.knative.dev/broker")),
```

Comment:
Can you confirm that omitting the label uses the namespace of the subscription?

Reply:
I confirmed this

Reply:
yep:
		c.EnqueueKey(types.NamespacedName{Namespace: object.GetNamespace(), Name: controllerKey})

controllerkey comes from the label. omitting the namespace means the line above gets executed, if you pass in something else besides "" it uses that to fetch the ns.